Source/Dafny/Compiler.cs

```diff
@@ -211,28 +179,48 @@ protected interface IClassWriter {
       if (idt == null) {
         return TypeArgumentInstantiation.ListFromClass(dt, typeArgs);
       } else {
-        Contract.Assert(typeArgs.Count == idt.TypeParametersUsedInConstructionByDefaultCtor.Length);
+        Contract.Assert(typeArgs.Count == idt.TypeParametersUsedInConstructionByGroundingCtor.Length);
         var r = new List<TypeArgumentInstantiation>();
         for (int i = 0; i < typeArgs.Count; i++) {
-          if (idt.TypeParametersUsedInConstructionByDefaultCtor[i]) {
+          if (idt.TypeParametersUsedInConstructionByGroundingCtor[i]) {
             r.Add(new TypeArgumentInstantiation(dt.TypeArgs[i], typeArgs[i]));
           }
         }
         return r;
       }
     }
 
+    protected bool NeedsTypeDescriptors(List<TypeArgumentInstantiation> typeArgs) {
+      Contract.Requires(typeArgs != null);
+      return typeArgs.Any(ta => NeedsTypeDescriptor(ta.Formal));
+    }
+
     protected virtual bool NeedsTypeDescriptor(TypeParameter tp) {
       Contract.Requires(tp != null);
       return tp.Characteristics.MustSupportZeroInitialization;
     }
 
-    protected virtual int EmitRuntimeTypeDescriptorsActuals(List<TypeArgumentInstantiation> typeArgs, Bpl.IToken tok, bool useAllTypeArgs, TargetWriter wr) {
+    protected abstract string TypeDescriptor(Type type, TextWriter wr, Bpl.IToken tok);
+
+    protected void EmitTypeDescriptorsActuals(List<TypeArgumentInstantiation> typeArgs, Bpl.IToken tok, TargetWriter wr, bool useAllTypeArgs = false) {
+      var prefix = "";
+      EmitTypeDescriptorsActuals(typeArgs, tok, wr, ref prefix, useAllTypeArgs);
+    }
```

Comment:
The method above could be used more often that it is. I recall seeing a number of instances of sep = ""; and then a call of EmitTypeDescriptorsActuals

Reply:
Apparently, all the other 16 uses make use of the `prefix` `ref`-parameter after the call.